Apps/Sandcastle/gallery/Camera Limiter.html

```diff
@@ -0,0 +1,195 @@
+var offsetForAxisAligned = 10000.0;
+var directionForAxisAligned = Cesium.Cartesian3.multiplyByScalar(centerForAxisAligned,
+    offsetForAxisAligned / Cesium.Cartesian3.magnitude(centerForAxisAligned),
+    new Cesium.Cartesian3());
+Cesium.Cartesian3.add(centerForAxisAligned, directionForAxisAligned, centerForAxisAligned);
+
+// The Axis Aligned Box
+var axisAligned = new Cesium.AxisAlignedBoundingBox(minForAxisAligned, maxForAxisAligned, centerForAxisAligned);
+
+// View the Axis Aligned Bounding Box
+var outlineAxisBox = viewer.entities.add({
+    name : 'Axis Aligned Box Outline',
+    position: centerForAxisAligned,
+    box : {
+        dimensions : new Cesium.Cartesian3(dimensionsForAxisAligned.x, dimensionsForAxisAligned.y, dimensionsForAxisAligned.z),
+        fill : false,
+        outline : true,
+        outlineColor : Cesium.Color.RED
+    }
+});
+
+
+
+
+// Used for OrientedBoundingBox Example
+var centerForOriented = Cesium.Cartesian3.fromDegrees(-90.0, 45.0, 45.0);
+var dimensionsForOriented = new Cesium.Cartesian3(300000.0, 300000.0, 500000.0);
+var halfDimensionsOriented = Cesium.Cartesian3.multiplyByScalar(dimensionsForOriented, 0.5, new Cesium.Cartesian3());
+
+var minForOriented = new Cesium.Cartesian3(-halfDimensionsOriented.x, -halfDimensionsOriented.x, -halfDimensionsOriented.x);
+var maxForOriented = new Cesium.Cartesian3(halfDimensionsOriented.x, halfDimensionsOriented.y, halfDimensionsOriented.z);
+
+// normalizing by dividing by magnitude and offsetting along direction vector so outside of earth surface
+var offsetForOriented = 10000.0;
+var directionForOriented = Cesium.Cartesian3.multiplyByScalar( centerForOriented,
+    offsetForOriented / Cesium.Cartesian3.magnitude(centerForOriented),
+    new Cesium.Cartesian3());
+Cesium.Cartesian3.add(centerForOriented, directionForOriented, centerForOriented);
+
+// The Oriented Box
+var oriented = new Cesium.AxisAlignedBoundingBox(minForOriented, maxForOriented, centerForOriented);
+
+// View the Oriented Bounding Box
+var outlineOrientedBox = viewer.entities.add({
+    name : 'Axis Aligned Box Outline',
+    position: centerForOriented,
+    box : {
+        dimensions : new Cesium.Cartesian3(dimensionsForOriented.x, dimensionsForOriented.y, dimensionsForOriented.z),
+        fill : false,
+        outline : true,
+        outlineColor : Cesium.Color.BLUE
+    }
+});
+
+
+
+// first initialization of the cameraLimiter
+viewer.camera.cameraLimiter = new Cesium.CameraLimiter({
+    boundingObject: boundingSphere
+});
+var saveCameraLimiter = new Cesium.CameraLimiter();
+var limiterDefined = true;
+
+// options for drop down
+var options = [{
+    text : 'BoundingSphere',
+    onselect : function() {
+        if (limiterDefined) {
+            viewer.camera.cameraLimiter.boundingObject = boundingSphere;
+        } else {
+            saveCameraLimiter.boundingObject = boundingSphere;
+        }
```

Comment:
Why not just create one instance of a `cameraLimiter`, and set `viewer.camera.cameraLimiter` to that or undefined when enabling/disabling?

Reply:
this is so that when the user clicks on the drop down while the limiter is not on, when the limiter box is rechecked again - the bounding object the limiter is currently using actually reflects what is currently selected in the drop down

Reply:
Doesn't this work?
```
var limtiter = new CameraLimiter(...);
```
```
onselect : function() {
    limiter.boundingObject = boundingSphere;
}
```
```
Sandcastle.addToggleButton('Limiter On', viewer.camera.cameraLimiter, function(checked) {
    if (checked) {
        viewer.camera.cameraLimiter = limiter;
    } else {
        viewer.camera.cameraLimiter = undefined;
    }
});
````

Reply:
that's partially the same idea of how i was implementing it - i can switch to that way for readability